docs/sql/tutorials/slack-chatbot.mdx

```diff
@@ -0,0 +1,281 @@
+
+<Tip>
+If you already have the API token of a Slack Bot, feel free to skip the following steps.
+</Tip>
+
+
+If you have a Slack Account, you have to generate the API token, please follow below instructions:
+
+<Accordion title="How to Generate API Keys" icon="slack" iconType="thin">
+
+    Here are the steps to generate the Slack API token:
+
+    1. Follow this [Link](https://api.slack.com/apps) and sign in with your Slack Account
+    2. Create a new `App` or select an existing `App`.
+    3. In the app settings, go to the `OAuth & Permissions` section.
+    4. Under the `Bot Token Scopes` section, add the necessary scopes for your application. You can add more later as well.
+    5. Install the bot to your workspace.
+    6. In the `OAuth Tokens & Redirect URLs` Section, copy the the `Bot User OAuth Access Token` (This is the API token which we need).
+    7. Open your Slack, in order to use the bot which we created, we have to add the bot into the channel where we want to use this.
+        - Go to the channel where you want to use the bot.
+        - Right Click on the `channel` and select `View Channel Details`.
+        - Select `Integrations`.
+        - Click on `Add an App`.
+        - You can see the name of the bot under the `In your workspace` Section, Go ahead and add the app to the channel.
+
+    Now, we can use the token from step 6 to initialize the Slack Handler in MindsDB.
+</Accordion>
+
+Now, let's establish a connection to our Slack using MindsDB SQL Editor:
+
+```sql
+CREATE DATABASE mindsdb_slack 
+WITH 
+    ENGINE = 'slack', 
+    PARAMETERS = { 
+        "token": "<slack-bot-token>" 
+    };
+```
+
+<Note>
+Please change the `slack-bot-token` with the token mentioned in `Bot User OAuth Access Token`.
+</Note>
+
+## Usage
+
+This query will create a database called `mindsdb_slack` and `channels` table automatically.
+
+Here is how to retrieve the 10 latest messages from the channel:
+
+```sql
+SELECT *
+FROM mindsdb_slack.channels
+WHERE channel="<channel-name>"
+LIMIT 10;
+```
+
+<Note>
+Please change the `channel-name` in the `WHERE` clause to the channel where, you added the bot in your Slack Workspace.
+</Note>
+
+However, you can also retrieve messages in alphabetical order by using:
+
+```sql
+SELECT *
+FROM mindsdb_slack.channels
+WHERE channel="<channel-name>"
+ORDER BY messages ASC
+LIMIT 5;
+```
+
+By default, it retrieves by the order the messages were sent, unless specified ascending/descending.
+
+Here is how to post messages:
+
+```sql
+INSERT INTO mindsdb_slack.channels (channel, message)
+VALUES
+    ("<channel-name>", "Hey MindsDB, Thanks to you! Now I can respond to my Slack messages through SQL Queries. 🚀 "),
+    ("<channel-name>", "It's never been that easy to build ML apps using MindsDB!");
+```
+
+Whoops! Sent it by mistake, No worries, use this to delete a specific message
+
+```sql
+DELETE FROM mindsdb_slack.channels
```

Comment:
The DELETE statement returns an error in the cloud editor:
```
[slack/mindsdb_slack]: 'Delete' object has no attribute 'columns'
```

Reply:
@tcchawla 
The DELETE statement results in an error: https://www.loom.com/share/693a8f0870784744944f99c9125d314c